jessica_hart/Assignments/OOP/bank_account.rb

```diff
@@ -0,0 +1,75 @@
+=begin
+Create a BankAccount class with the following:
+- Method that returns a user's account number, account number should be generated by a private method, account number should be random
+- Method that returns the user's checking account balance
+- Method that returns the user's saving account balance
+- Allow a user to deposit money into either their checking or saving account
+- Allow a user to withdraw money from one of their accounts, return an error if there are insufficient funds
+- Allow the user to view the total amount of money they have at the bank
+- Track how many accounts the bank currently has
+- Add an interest_rate attribute that is not accessible by the user. Set it to 0.01
+- Method called account_information that displays the users account number, total money, checking account balance, saving account balance and interest rate
+- A user should not be able to set any attributes from the BankAccount class
+=end
+
+class BankAccount
+  attr_reader :account_number, :checkings_balance, :savings_balance
+  attr :interest_rate
+  @@total_accounts = 0
+  @interest_rate = 0.01
+
+  def initialize
+    generate_account_number
+    @@total_accounts += 1
+    @savings_balance = 0
+    @checkings_balance = 0
+  end
+
+  def deposit(account, amount)
```

Comment:
I'd suggest making checking and savings a constant and referencing it from there. so BankAccount::SAVINGS, it will prevent typos in the long run.